src/state/session/index.tsx

```diff
@@ -317,9 +297,16 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
         }
       })
     },
-    [setStateWrapped],
+    [setState],
   )
 
+  React.useEffect(() => {
+    persisted.write('session', {
```

Comment:
Note you probably don't want to do this after first render. So maybe:

- Keep a wrapper that sets a dirty flag in a ref
- In the Effect, if the dirty flag is true, persist and unset the flag.